vault/seal_autoseal_test.go

```diff
@@ -0,0 +1,151 @@
+package vault
+
+import (
+	"bytes"
+	"context"
+	"reflect"
+	"testing"
+
+	proto "github.com/golang/protobuf/proto"
+	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/hashicorp/vault/vault/seal"
+)
+
+type phy struct {
```

Comment:
Why not just use a regular inmem backend?

Reply:
It doesn't capture the previous versions for each entry which allows the test to verify that a call to `UpgradeKeys` only updates the values if the key has changed.

Reply:
I added a comment to `phy` to explain what it is used for.